### PR TITLE
Adding ConnectionIldeTimeout property on servicebusconnection to dete…

### PR DESCRIPTION
**Pull request by yvgopal**
Adding ConnectionIldeTimeout property on servicebusconnection to detect network errors sooner. In some cases, underlying transport layer is very slow to detect a connection drop and takes many minutes to throw a socket error. This timeout will close any connection that sees no traffic for the duration of the timeout.

### CODE REVIEW

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/ServiceBusConnection.cs [hunk 1]
```diff
@@ -132,6 +132,16 @@ internal ServiceBusConnection(RetryPolicy retryPolicy = null)
         /// <remarks>Defaults to 1 minute.</remarks>
         public TimeSpan OperationTimeout { get; set; }
 
+        /// <summary>
+        /// ConnectionIdleTimeout is applied when a connection recevies no traffic for a certain period of time.
+        /// In some situations, underlying transport layer takes really long time to detect a socket error and it may
+        /// cause some operations to time out after a long wait. This timeout helps to detect and close an idle connection
+        /// without waiting for the network layer to identify a socket error. But this setting may result in some incrased
```
Comment (DorothySun216) [inline sdk/servicebus/Microsoft.Azure.ServiceBus/src/ServiceBusConnection.cs:139]: nit: increased

**Review by DorothySun216 — APPROVED** [review]

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/ServiceBusConnection.cs (hunk 1 above)
Comment (SeanFeldman) [inline sdk/servicebus/Microsoft.Azure.ServiceBus/src/ServiceBusConnection.cs:139]: @DorothySun216 it's not a nit but a good suggestion. I took it a step forward 🙂

```suggestion
        /// without waiting for the network layer to identify a socket error. But this setting may result in some increased
```

**Comment by jsquire** [comment]
Hi @yvgopal.  There hasn't been recent engagement on this PR.  Would you please be so kind as to let us know if this is still an active work stream or if the PR should be closed out?

**Comment by jsquire** [comment]
Since there hasn't been recent engagement, I'm going to close this out.  Please feel free to reopen if you'd like to continue working on these changes.

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/ServiceBusConnection.cs [hunk 2]
```diff
@@ -132,6 +132,16 @@ internal ServiceBusConnection(RetryPolicy retryPolicy = null)
         /// <remarks>Defaults to 1 minute.</remarks>
```
Comment (yvgopal) [inline sdk/servicebus/Microsoft.Azure.ServiceBus/src/ServiceBusConnection.cs:132]: /check-enforcer reset

**Pull request closed without merging**